### PR TITLE
ERC721 template contract for bridgeless minting

### PR DESCRIPTION
**Pull request by zorjak**
- Solidity ERC721 template contract designed for bridgeless minting. 
- Enable ts linter
- Script for contract deployment
- Readme and License

### CODE REVIEW

**Review by AlexRipoll — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -54,7 +53,6 @@ Make sure to obtain the corresponding API_KEY, and set it in the `.hardhat.confi
 
 ## Additional resources
 
-* **LAOS Whitepaper**:  For a comprehensive understanding of the LAOS project, you can review the [whitepaper](https://github.com/freeverseio/laos-whitepaper)
-
-* **LAOS Roadmap**: To explore the future plans and updates for LAOS, visit the [roadmap](https://github.com/freeverseio/laos-roadmap)
+- **LAOS Whitepaper**: For a comprehensive understanding of the LAOS project, you can review the [whitepaper](https://github.com/freeverseio/laos-whitepaper)
```
Comment (tonimateos) [inline README.md:56]: try not to use "you" in formal language: "For a comprehensive understanding of the LAOS project, please review the..."

File: README.md [hunk 2]
```diff
@@ -40,11 +41,9 @@ To deploy the contract, edit the `hardhat.config.ts` file, and use the following
 npx hardhat run --network <network-name> scripts/deploy.ts
 ```
 
-
 ### Verifying Contract
 
-The following command can be used to 
-verify the deployed contracts on [etherscan](https://etherscan.io/) or any other EVM scan (e.g [Polygonscan](https://polygonscan.com/)):
+The following command can be used toverify the deployed contracts on [etherscan](https://etherscan.io/) or any other EVM scan (e.g [Polygonscan](https://polygonscan.com/)):
```
[the remark this answers is not in the document]
Comment (zorjak) [reply README.md]: ok

File: README.md (hunk 1 above)
Comment (zorjak) [reply README.md]: ok

File: contracts/ERC721BridgelessMinting.sol [hunk 3]
```diff
@@ -7,19 +7,19 @@ import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
  * @title Contract for bridgeless minting of ERC721 tokens
  * @author Freeverse.io, www.freeverse.io
  * @dev The contract is an extension of OpenZeppelin ERC721
- * @dev The contract allocates 2^96 slots to every possible 160b address,
- * @dev to be filled in the Evolution consensus system
- * @dev The contract has possibility of adding default operator that can change token ownership
+ * @dev The contract allocates 2^96 slots to every possible 160b address, to be filled in the Evolution consensus system
```
[the remark this answers is not in the document]
Comment (zorjak) [reply contracts/ERC721BridgelessMinting.sol]: ok

File: contracts/ERC721BridgelessMinting.sol [hunk 4]
```diff
@@ -81,7 +82,7 @@ contract ERC721BridgelessMinting is ERC721 {
 
     /**
      * @notice Returns the current owner of a token
-     * @dev The function overrides the one in the base ERC721 contract, and it returns the token's correct owner if the token has never been previously transferred.
+     * @dev @dev This function overrides the one in the base ERC721 contract. On deploy, all tokens have an assigned owner, encoded in their tokenId, and determined via usage of initOwner(tokenId). Upon transfer, the new owner is stored in, and retrieved from, the contract storage.
```
Comment (tonimateos) [inline contracts/ERC721BridgelessMinting.sol:85]: is this the way very long comment lines work? Don't you need to split it into multiple lines following the solidity convention? check: https://www.bitdegree.org/learn/solidity-syntax

**Review by tonimateos — CHANGES_REQUESTED** [review]

File: contracts/ERC721BridgelessMinting.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (zorjak) [reply contracts/ERC721BridgelessMinting.sol:85]: ok

File: contracts/ERC721BridgelessMinting.sol [hunk 5]
```diff
@@ -7,9 +7,13 @@ import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
...
  * @dev The contract is an extension of OpenZeppelin ERC721
- * @dev The contract allocates 2^96 slots to every possible 160b address, to be filled in the Evolution consensus system except for the null address, which cannot own any slot
- * @dev the null address is used only as the implicit target address when executing the burn method"
- * @dev The contract has the possibility of adding default operator that can chang token ownership
+ * @dev The contract allocates 2^96 slots to every possible 160b address, to be
+ *  filled in the Evolution consensus system except for the null address, which
+ *  cannot own any slot
+ * @dev the null address is used only as the implicit target address when
+ *  executing the burn method"
+ * @dev The contract has the possibility of adding default operator that can
```
Comment (tonimateos) [inline contracts/ERC721BridgelessMinting.sol:15]: This is not true anymore

File: contracts/ERC721BridgelessMinting.sol [hunk 6]
```diff
@@ -61,8 +67,10 @@ contract ERC721BridgelessMinting is ERC721 {
 
     /**
      * @notice Returns the initial owner address which is encoded in the tokenId
-     * @dev This function returns the same value regardless of whether the token has been transferred once or more times.
-     * @dev Use ownerOf() to query current owner of an token, as opposed to the init owner.
+     * @dev This function returns the same value regardless of whether the
+     * token has been transferred once or more times.
+     * @dev Use ownerOf() to query current owner of an token, as opposed to the
+     *  init owner.
```
Comment (tonimateos) [inline contracts/ERC721BridgelessMinting.sol:73]: the init owner ARE encoded -> IS encoded

**Review by tonimateos — CHANGES_REQUESTED** [review]

File: contracts/ERC721BridgelessMinting.sol [hunk 7]
```diff
@@ -12,7 +12,6 @@ import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
  *  cannot own any slot
  * @dev the null address is used only as the implicit target address when
  *  executing the burn method"
- * @dev The contract has the possibility of adding default operator that can
```
Comment (tonimateos) [inline contracts/ERC721BridgelessMinting.sol]: these lines are now a mess... with a line extra should be removed, there is a " at the end of: with method", there is a typo chang -> changed...

**Review by tonimateos — CHANGES_REQUESTED** [review]

File: contracts/ERC721BridgelessMinting.sol [hunk 8]
```diff
@@ -12,7 +12,6 @@ import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
  *  cannot own any slot
  * @dev the null address is used only as the implicit target address when
  *  executing the burn method"
- *  chang token ownership
```
Comment (tonimateos) [inline contracts/ERC721BridgelessMinting.sol]: the previous line, as said in the previous PR, has typo at the end: "

**Review by tonimateos — CHANGES_REQUESTED** [review]

**Review by tonimateos — APPROVED** [review]

**Pull request merged**